src/vscode-dts/vscode.d.ts

```diff
@@ -11229,6 +11229,12 @@ declare module 'vscode' {
 		 * Is `undefined` if the reason is not known.
 		*/
 		readonly reason: TextDocumentChangeReason | undefined;
+
+		/**
+		 * Optional identifier of the top-most intent that triggered this event e.g. all editing operations
+		 * triggered by user pasting text from clipboard will have "paste" source.
+		 */
+		readonly source?: string | undefined;
```

Comment:
We don't accept direct changes to the API, they must go through a proposal-process in which they are examined and discussed. See https://github.com/Microsoft/vscode/wiki/Extension-API-process for details. 

For this particular case I see a conflict with `TextDocumentChangeReason` a few lines above. I believe it serves a similar/same intent